### DISCUSSION TITLE
How to implement whole site scheduled backup?

### DISCUSSION THREAD

**Discussion by xperiandri** [OrchardCMS/OrchardCore#9359] [Q&A]
Currently, we use a deployment plan to export all the items.
Can it be automated?

**Comment by weirdyang** [comment]
I think you can either create a module to perform a background task, or have a workflow with a timer to execute this on a schedule.

**Comment by Piedone** [comment]
An approach safer than relying on deployments is to use the tools of the platform you're running on to take a snapshot. You need to back up the DB, Media storage, and depending on your app's configuration and Orchard Core version, also the App_Data folder (since e.g. Lucene indices, data protection keyrings, content definitions may be saved there). This guarantees that 100% of your site's data is backed up - except for the binaries but that you can of course reproduce any time from source.

E.g. in Azure all this can be handled by built-in services in some way.

**Reply by mariojsnunes** [reply]
Just to add that you shouldn't need to backup the app_data folder if you configured OrchardCore.DataProtection.Azure and OrchardCore.Media.Azure.
Lucene indexes are rebuilt automatically on startup if they are missing (https://github.com/OrchardCMS/OrchardCore/pull/9166)
All you should need to backup is the databases.

**Reply by jptissot** [reply]
We use Kubernetes to host our OC instance and we use https://velero.io/ for backups.

**Reply by Piedone** [reply]
Hence I said "depending on your app's configuration and Orchard Core version" ;). You may not need to back up App_Data indeed.
